src/main/java/org/folio/linked/data/validation/spec/impl/SpecProviderImpl.java

```diff
@@ -0,0 +1,58 @@
+package org.folio.linked.data.validation.spec.impl;
+
+import static org.folio.linked.data.configuration.CacheConfiguration.SPEC_RULES;
+
+import feign.FeignException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.folio.linked.data.client.SpecClient;
+import org.folio.linked.data.validation.spec.SpecProvider;
+import org.folio.rspec.domain.dto.SpecificationDto;
+import org.folio.rspec.domain.dto.SpecificationDtoCollection;
+import org.folio.rspec.domain.dto.SpecificationRuleDto;
+import org.folio.rspec.domain.dto.SpecificationRuleDtoCollection;
+import org.springframework.cache.annotation.CacheEvict;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.http.ResponseEntity;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class SpecProviderImpl implements SpecProvider {
+
+  private final SpecClient client;
+
+  @Cacheable(value = SPEC_RULES)
```

Comment:
If we are going with this option, then we can do  `.filter(rule -> rule.getCode().equals("invalidLccnSubfieldValue"))` here so that we are caching only the required rule.

(This may not be possible if we implement the suggestion that @PBobylev  provided. Not a big deal as the number of rules is not that big)